### PR TITLE
sdhc: sdhc_spi: compile-time determine if `sdhc_ones` needed

### PR DESCRIPTION
**Pull request by JordanYates**
Move the `overrun-character` property from the common Nordic SPI binding to the `spi-controller` base binding. This gives users of the SPI interface a way to query what the default value is at compile-time, and potentially avoid allocation of large constant buffers.

Expose the new common property `overrun-character` from the devicetree nodes, falling back to the `SPI_MOSI_OVERRUN_UNKNOWN` value.

Use the new `SPI_MOSI_OVERRUN_DT` macro to determine at compile-time whether the 512 byte array of `sdhc_ones` is required.

This PR does not attempt to peruse datasheets to determine the default MOSI values clocked out by the dozens of SPI drivers in tree, only to make it trivial for contributors to expose that information in the future.

### CODE REVIEW

**Review by danieldegrasse — COMMENTED** [review]
I wonder if it would make more sense to support setting this property at runtime? Just musing here, but is this a fixed value on the nordic controller or a configurable setting? If it is configurable perhaps we make it possible to set that at runtime.

If we want to keep this as a devicetree property though, than I am ok with this approach from the SDHC side

**Comment by JordanYates** [comment]
> I wonder if it would make more sense to support setting this property at runtime? Just musing here, but is this a fixed value on the nordic controller or a configurable setting? If it is configurable perhaps we make it possible to set that at runtime.

It is a configurable setting but currently it is a constant from devicetree. I'm not sure what value could be derived from making it configurable at runtime. It doesn't seem like a common option among SoC vendors, and if you want a specific value you can always do what was done here with `sdhc_ones`.

The value as a compile-time constant is that you can do what this PR does and optimize out large constant arrays if the value matches the value you want.

In short, if it is a runtime parameter, this PR is impossible.

**Review by anangl — APPROVED** [review]

**Comment by danieldegrasse** [comment]
> The value as a compile-time constant is that you can do what this PR does and optimize out large constant arrays if the value matches the value you want.

> In short, if it is a runtime parameter, this PR is impossible.

Yes, this is a good point. Let's keep it as devicetree then

**Review by danieldegrasse — APPROVED** [review]
Looks good from SDHC side

**Review by tbursztyka — APPROVED** [review]
lgtm
